firewood/src/merkle/node.rs

```diff
@@ -305,6 +297,17 @@ impl Node {
     }
 }
 
+#[repr(C)]
+struct Meta {
+    root_hash: [u8; TRIE_HASH_LEN],
+    attrs: NodeAttributes,
+    is_encoded_longer_than_hash_len: Option<bool>,
+}
```

Comment:
This is 100% stack-based and can actually be (de)serialized in one shot. I didn't do that here though, that can be done later if necessary.